pkgs/applications/misc/prusa-slicer/bambu-studio.nix

```diff
@@ -0,0 +1,88 @@
+{ lib, fetchFromGitHub, fetchpatch, prusa-slicer
+, curl
+, wxGTK31
+, glfw
+, glib-networking
+, mesa
+, webkitgtk
+}:
+let
+  wxGTK31' = wxGTK31.overrideAttrs (old: {
+    configureFlags = old.configureFlags ++ [
+      # Disable noisy debug dialogs
+      "--enable-debug=no"
+    ];
+  });
+
+  versions = {
+    official = rec {
+      version = "unstable-2022-12-16";
+
+      src = fetchFromGitHub {
+        owner = "bambulab";
+        repo = "BambuStudio";
+        rev = "73679f6f2e1111cfeefba146f2897d2d593ae452";
+        hash = "sha256-j66FE81S7GNg7KyiBbLc5RvfHIkBAcxDQvQ99/ipzKM=";
+      };
+
+      patches = [
+        # Create the data dir if it doesn't exist
+        (fetchpatch {
+          url = "https://github.com/zhaofengli/BambuStudio/commit/a722c53f83f78775144ef6edd4b9debffff05cf5.patch";
+          hash = "sha256-wytf7Uuynh282yd4F35W1w7qAw/Uz1dAW80WGkxxs00=";
+        })
+
+        # Fix for webkitgtk linking
+        ./0001-not-for-upstream-CMakeLists-Link-against-webkit2gtk-.patch
+      ];
+    };
+
+    # TODO: SoftFever
+  };
+
+  override = { version, src, patches ? [], ... } @ args: super: rec {
+    pname = "bambu-studio";
+    inherit version src patches;
+
+    buildInputs = super.buildInputs ++ [
+      glfw
+      glib-networking
+      mesa.osmesa
+      webkitgtk
+    ];
+
+    prePatch = ''
+      # Since version 2.5.0 of nlopt we need to link to libnlopt, as libnlopt_cxx
+      # now seems to be integrated into the main lib.
+      sed -i 's|nlopt_cxx|nlopt|g' cmake/modules/FindNLopt.cmake
+    '';
+
+    cmakeFlags = super.cmakeFlags ++ [
+      "-DBBL_RELEASE_TO_PUBLIC=1"
+      "-DBBL_INTERNAL_TESTING=0"
+      "-DDEP_WX_GTK3=ON"
+      "-DSLIC3R_BUILD_TESTS=0"
+      "-DCMAKE_CXX_FLAGS=-DBOOST_LOG_DYN_LINK"
+    ];
+
+    postInstall = null;
+
+    passthru = allVersions;
+
+    meta = with lib; {
+      description = "PC Software for BambuLab's 3D printers";
+      homepage = "https://github.com/bambulab/BambuStudio";
+      license = licenses.agpl3;
+      maintainers = with maintainers; [ zhaofengli ];
+      mainProgram = "bambu-studio";
+    };
+  } // args;
+
+  prusa-slicer' = prusa-slicer.override {
```

Comment:
@cab404 who maintains super-slicer and me discussed somewhere (I can't find it ATM) that it makes sense to not use overrides for prusa-slicer anymore as there are quite a lot of differences now, i.e. super-slicer might be moved to an independent package, even if it involves some duplication. From the length of this file it looks like that might be a good idea for Bambo-Studio. What do you think?

Reply:
Yeah, I agree and the build stage overrides indeed look very fragile. I'll rework this in a bit.

Reply:
I've refactored the package to no longer be an override.